apache/superset
# feat: Migrates Pivot Table v1 to v2

### SUMMARY
This PR migrates the Pivot Table v1 chart to v2. In detail, it does the following:
- Creates a processor for the Pivot Table chart to convert v1 `form_data` to v2
- Augments the base `MigrateViz` class to support migrating generic chart axes related fields given that some legacy plugins only work without generic chart axes
- Creates a database migration that invokes the processor
- Creates a migration test
- Removes the legacy Pivot Table code and unregisters the plugin.
- Removes old plugin references throughout the codebase
- Updates unit tests
- Updates E2E tests
- Updates the example dashboards
- Updates translation files

The following table summarizes the differences in the `form_data` structure between the versions:

v1 | v2 | Notes
-- | -- | --
columns | groupbyColumns |  
combine_metric | combineMetric |  
granularity_sqla | adhoc_filters.subject | Only if GENERIC_CHART_AXES is enabled
groupby | groupbyRows |  
number_format | valueFormat |  
pandas_aggfunc | aggregateFunction | Values changed between versions
pivot_margins | rowTotals, colTotals |  
row_limit | series_limit |  
time_range | adhoc_filters.comparator | Only if GENERIC_CHART_AXES is enabled
timeseries_limit_metric | series_limit_metric |  
transpose_pivot | transposePivot |  

\* All other keys are the same or exist only in the target version
\* The time column type (simple or SQL) affects the `adhoc_filters` value

### AFTER SCREENSHOT
<img width="1065" alt="Screenshot 2023-04-19 at 10 24 26" src="https://user-images.githubusercontent.com/70410625/233088808-1c079eb9-5d62-4dab-89da-ad02236a694a.png">

### TESTING INSTRUCTIONS
1 - Make sure all Pivot Table (legacy) charts were converted to Pivot Table
2 - Make sure Pivot Table (legacy) is not available anymore in the viz picker
3 - Make sure that it's possible to revert the migration by executing `superset db downgrade` and reverting this PR

### ADDITIONAL INFORMATION
